schema-registry/authorizer/authorizer-core/src/main/java/com/hortonworks/registries/schemaregistry/authorizer/core/Authorizer.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2016-2019 Cloudera, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.registries.schemaregistry.authorizer.core;
+
+import java.util.Map;
+import java.util.Set;
+
+public interface Authorizer {
+
+    void configure(Map<String, Object> props);
+
+    boolean authorize(Resource resource, AccessType accessType, UserAndGroups userAndGroups);
+
+
+    enum AccessType {
+        CREATE("create"),
+        READ("read"),
+        UPDATE("update"),
+        DELETE("delete");
+
+        private final String name;
+
+        AccessType(String name) { this.name = name; }
+
+        public String getName() { return name; }
+    }
+
+    class UserAndGroups {
+        private String user;
+        private Set<String> groups;
+
+        public UserAndGroups(String user, Set<String> groups) {
+            this.user = user;
+            this.groups = groups;
+        }
+
+        public String getUser() {
+            return user;
+        }
+
+        public Set<String> getGroups() {
+            return groups;
+        }
+    }
```

Comment:
They are important because they are used to form exception messages for AuthorizationException. They helped me to find a couple of minor bugs. So let's leave them.

Reply:
Done!